ale_linters/kotlin/kotlinc.vim

```diff
@@ -46,8 +46,9 @@ function! ale_linters#kotlin#kotlinc#GetCommand(buffer, import_paths) abort
 
     " If the config file is enabled and readable, source it
     if ale#Var(a:buffer, 'kotlin_kotlinc_enable_config')
-        if filereadable(expand(ale#Var(a:buffer, 'kotlin_kotlinc_config_file'), 1))
-            execute 'source ' . ale#Escape(expand(ale#Var(a:buffer, 'kotlin_kotlinc_config_file'), 1))
+        let l:conffile = fnameescape(expand(ale#Var(a:buffer, 'kotlin_kotlinc_config_file'), 1))
+        if filereadable(l:conffile)
```

Comment:
You have to apply `filereadable` to the filename which has not yet been escaped. It's best to escape strings right where you use them. I'll sort this out on master.